torch/onnx/symbolic_opset12.py

```diff
@@ -209,3 +209,48 @@ def unfold(g, input, dimension, size, step):
         return squeeze
     else:
         return _unimplemented("Unfold", "input size not accessible")
+
+@parse_args('v', 'v', 'is', 'is', 'v')
+def tensordot(g, input_a, input_b, dims_a, dims_b, out=None):
+    if out is not None:
+        _unimplemented("Tensordot", "Out parameter is not supported for tensordot.")
+
+    from torch.onnx.symbolic_opset9 import view, permute
+    for i in range(len(dims_a)):
+        if sym_help._get_tensor_dim_size(input_a, dims_a[i]) != sym_help._get_tensor_dim_size(input_b, dims_b[i]):
+            raise RuntimeError("Dimensions are not matched for operator tensordot: a - {}, b - {}.".format(dims_a, dims_b))
+
+    dim_count_a = input_a.type().dim()
+    dim_count_b = input_b.type().dim()
+    for i in range(len(dims_a)):
+        if (dims_a[i] < 0):
+            dims_a[i] += dim_count_a
+    for i in range(len(dims_b)):
+        if (dims_b[i] < 0):
+            dims_b[i] += dim_count_b
+
+    left_dims_a = [i
+                   for i in range(dim_count_a)
+                   if (i not in dims_a)]
+    left_dims_b = [i
+                   for i in range(dim_count_b)
+                   if (i not in dims_b)]
+
+    left_dims_sizes_a = [sym_help._get_tensor_dim_size(input_a, left_dims_a[i])
+                         for i in range(len(left_dims_a))]
+    left_dims_sizes_b = [sym_help._get_tensor_dim_size(input_b, left_dims_b[i])
+                         for i in range(len(left_dims_b))]
+
+    new_input_a = permute(g, input_a, left_dims_a + dims_a)
+    new_input_b = permute(g, input_b, dims_b + left_dims_b)
+
+    output_a = view(g, new_input_a, left_dims_sizes_a + [-1])
+    output_a = view(g, output_a, [-1] + [sym_help._get_tensor_dim_size(output_a, len(left_dims_sizes_a))])
```

Comment:
Can we directly use `view` once? Feel like the first `view` is only used to calculate the dim size.

Reply:
In my latest commit, view() has been removed. Please help to take another review. Thanks